.github/workflows/changelog.yml

```diff
@@ -11,15 +11,18 @@ on:
       - CHANGELOG.md
   pull_request_target:
 
+env:
+  IN_MAINTAINER_LIST: ${{ contains(secrets.MAINTAINER_LIST, github.actor) }}
+
 jobs:
   changes:
-    if: github.event_name == 'pull_request_target' && !contains( secrets.MAINTAINER_LIST, github.actor)
     name: Filter Changes
     runs-on: ubuntu-latest
     outputs:
       changed: ${{ steps.filter.outputs.changed }}
     steps:
       - uses: dorny/paths-filter@v2
+        if: github.event_name == 'pull_request_target' && env.IN_MAINTAINER_LIST == 'false'
```

Comment:
I'm not familiar with the data type that gets returned in `IN_MATAINER_LIST` 😅  but if the equality check works here as a String, LGTM

Reply:
Mainly commenting in case anyone ever needs to reference this.

This is what worked in my testing, but just to back myself up, I checked back into the documentation and found [this example](https://docs.github.com/en/actions/learn-github-actions/expressions#example-returning-a-json-data-type). From what I can tell, even if the environment evaluates to a `bool`, when you reference it, it'll be a `string`. In order to use it as a `bool` (so that we could do `!env.IN_MAINTAINER_LIST`), we'd need to use `!fromJSON(env.IN_MAINTAINER_LIST)`, or something like that -- I didn't test this route since this seemed to work just as well.